src/main/java/com/dnd/runus/application/running/RunningRecordService.java

```diff
@@ -101,6 +108,45 @@ public RunningRecordSummaryResponse getRunningRecordSummaries(long memberId, Loc
         return RunningRecordSummaryResponse.from(records);
     }
 
+    @Transactional(readOnly = true)
+    public RunningRecordWeeklySummaryResponse getWeeklySummary(
+            long memberId, RunningRecordWeeklySummaryType summaryType) {
+
+        OffsetDateTime today = LocalDate.now().atStartOfDay().atOffset(defaultZoneOffset);
+        OffsetDateTime startWeekDate = today.with(DayOfWeek.MONDAY);
+        OffsetDateTime nextOfEndWeekDate = startWeekDate.plusWeeks(1);
+
+        List<DailyRunningRecordSummary> weekSummaries;
+        int avgValue;
+        double conversionFactor;
+
+        if (summaryType.equals(RunningRecordWeeklySummaryType.DISTANCE)) {
+            weekSummaries = runningRecordRepository.findDailyDistancesMeterByDateRange(
+                    memberId, startWeekDate, nextOfEndWeekDate);
+            avgValue = runningRecordRepository.findAvgDistanceMeterByMemberIdAndDateRange(
+                    memberId, startWeekDate, nextOfEndWeekDate);
+            conversionFactor = METERS_IN_A_KILOMETER;
+        } else {
+            weekSummaries = runningRecordRepository.findDailyDurationsSecByDateRange(
+                    memberId, startWeekDate, nextOfEndWeekDate);
+            avgValue = runningRecordRepository.findAvgDurationSecByMemberIdAndDateRange(
+                    memberId, startWeekDate, nextOfEndWeekDate);
+            conversionFactor = SECONDS_PER_HOUR;
+        }
+
+        double[] weeklyValues = new double[7];
+        for (DailyRunningRecordSummary Summary : weekSummaries) {
+            OffsetDateTime dateTime = Summary.date().atStartOfDay().atOffset(defaultZoneOffset);
+            weeklyValues[dateTime.get(DAY_OF_WEEK) - 1] = Summary.sumValue() / conversionFactor;
+        }
```

Comment:
`summary.date()`가 LocalDate라서, 아래처럼 바로 dayOfWeek를 구하면 좋을 것 같아요

```suggestion
        double[] weeklyValues = new double[7];
        for (DailyRunningRecordSummary summary : weekSummaries) {
            int dayOfWeek = summary.date().getDayOfWeek().getValue() - 1;
            weeklyValues[dayOfWeek] = summary.sumValue() / conversionFactor;
        }
```

Reply:
`summary.date()` 가 db에서 조회 해올 때, `cast(RUNNING_RECORD.START_AT, SQLDataType.DATE)` 이렇게 조회해와서 시간대의 정보가 포함되어 있지 않아요!

추 후에 사용자를 글로벌하게 받게 된다면 이 부분에 대해서 고려해서 리팩터링을 진행해야 될 것 같아요!